### DISCUSSION TITLE
Disk and Memory Cache... When and How to Use It

### DISCUSSION THREAD

**Discussion by ghuDaYuYu** [ghDaYuYu/foo_discogs#119] [Feature Request]
Is it [worth](https://github.com/ghDaYuYu/foo_discogger/issues/111) it? It delivers rapid responses and sometimes ensures data availability.
But also stands in the way of new releases or keeping updated with recent Discogs catalog edits.
Disk Cache also uses a significant amount of space.

**Comment by bassstorm** [comment]
For me it is simple: tagging is a one-time task mostly. If that happens I need to re-tag previously tagged files, for sure I'd like to get up-to-date, fresh data, no matter what response time is. Otherwise (besides short lifetime of temporary cache) it makes no practical sense, i.e. there's official [monthly dumps](https://discogs-data-dumps.s3.us-west-2.amazonaws.com/index.html) available after all :)

**Reply by ghuDaYuYu** [reply]
I agree, good reasons to disable cache totally, response time does not matter and re-tag to get the latest data.
I acknowledged the 'otherwise-besides', although, for the sake of simplicity, in case someone might find it useful...

**Disabling cache**

When:
- Response time does not matter.
- Data availability, i.e. network and server accessibility do not matter.

How:
- Set the Cache memory 'Max objects' option value to 0.
- Disable Disk cache 'Read' and 'Write' options.
- Optionally, remove folder `<profile>\configuration\foo_discogger-cache`

**Reply by bassstorm** [reply]
Personally, I ended up with Cache Expiration set to 1 day. For me that makes sense, because my regular use case is to tag a handful of new-to-my-library releases from an artist in one go, so there's obviously close to zero chances that I miss an important update within half an hour or so, while it adds convenience outlined in original post, e.g. no need to query whole artist discography again and fast response time in general.

**Reply by ghuDaYuYu** [reply]
> Disk Cache also uses a significant amount of space

In case it would be useful, here is a draft Python script to delete expired entries... [deeDiscoggerCache.py](https://gist.github.com/ghuDaYuYu/0860f6a6f90a71a8b0e5c2922f8deb14).

**Reply by bassstorm** [reply]
Just out of fellow developer's curiosity, why script instead of making it part of the component? :)

**Comment by ghuDaYuYu** [comment]
Not sure if this answers the developer aspect.
I have considered 'Clean disk cache' and 'Delete expired objects' added to the Cache preferences panel and may give it a try in future releases.
The current approach is to roughly split data age and storage into separate solution domains.
Related to your question, the second is cost-effective and adaptable.

Weekly free up drive space by Andrew Smolko' foo_scheduler...


![deeDiscoggerCache](https://github.com/user-attachments/assets/03fdf842-29f0-4c7a-9bb0-8a957d3cbd94)


... advanced users can easily upgrade the [script](https://gist.github.com/ghuDaYuYu/0860f6a6f90a71a8b0e5c2922f8deb14), adding new features like artist exclusion lists, expiration ranges, etc.

**Reply by bassstorm** [reply]
Mmm, that's very interesting, didn't know about foo_scheduler. Foobar2000 ecosystem still can positively surprise me :)
